src/server/json_family.cc

```diff
@@ -259,8 +279,108 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
   return vec;
 }
 
+template <typename Op>
+OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, string_view path,
+                                    double num, Op arithmetic_op) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  bool is_result_overflow = false;
+  double int_part;
+  bool has_fractional_part = (modf(num, &int_part) != 0);
+  json output(json_array_arg);
+
+  auto cb = [&](const string& path, json& val) {
+    if (val.is_number()) {
+      double result = arithmetic_op(val.as<double>(), num);
+      if (isinf(result)) {
+        is_result_overflow = true;
```

Comment:
you can return early, no?

Reply:
Yes sure.
Moreover, I can't interrupt the JsonReplace operation(maybe by throwing an exception). Still, I can add to the first line of the JsonReplace lambda a check that verifies the overflow doesn't occur if so, to reduce CPU  utilization in case of significant input.

Reply:
i just meant, returning early from the lambda because I do not see a point in running the rest of its code.

Reply:
Done